### ISSUE TITLE
Proxy not working with multiprocessing as 'True'

### ISSUE THREAD

**Issue by rajvidhimar** [saltstack/salt#38864]
## Description of Issue/Question
I am running junos proxy. The way junos proxy is that, during initialization, a connection is made to the juniper device. This connection is then stored in an dictionary and this connection stored in the dictionary is used in the execution modules. 
There is an option to set multiprocessing in the proxy config file. By default, multiprocessing is set to True, which means every new command executed on the proxy minion, starts in a new process. When multiprocessing is set 'False', each command is executed in a different thread. 
The problem with junos proxy is that, when multiprocessing is set to 'True', we are unable to run the execution module as the connection used from the dictionary becomes unusable. The connection, object accessed in the execution module is the same as the one made in the initialization process. Still, the connection doesn't seem to work. Things work fine when multiprocessing is set to 'False', i.e., when threading is used instead of multiprocessing.

**Comment by mirceaulinic** [comment]
Yes, `multiprocessing: False` is mandatory for SSH-based proxy minions: https://docs.saltstack.com/en/latest/topics/proxyminion/ssh.html#proxy-minion-ssh-end-to-end-example.

**Comment by gtmanfred** [comment]
As @mirceaulinic `multiprocessing: False` is required for some proxy minions.

I am going to tag this as a documentation fix, to add this to the documentation for minions that need this set.

Thanks,
Daniel

**Comment by rajvidhimar** [comment]
There is an ongoing discussion regarding the same issue here: https://github.com/saltstack/salt/issues/36964
